keras/layers/convolutional.py

```diff
@@ -379,6 +379,95 @@ def get_config(self):
         return dict(list(base_config.items()) + list(config.items()))
 
 
+class TransposedConvolution2D(Convolution2D):
+    '''Transposed convolution operator for filtering windows of two-dimensional inputs.
+    When using this layer as the first layer in a model,
+    provide the keyword argument `input_shape`
+    (tuple of integers, does not include the sample axis),
+    e.g. `input_shape=(3, 128, 128)` for 128x128 RGB pictures.
+    '''
+    def __init__(self, nb_filter, nb_row, nb_col, output_shape,
+                 init='glorot_uniform', activation='linear', weights=None,
+                 border_mode='valid', subsample=(1, 1),
+                 dim_ordering=K.image_dim_ordering(),
+                 W_regularizer=None, b_regularizer=None, activity_regularizer=None,
+                 W_constraint=None, b_constraint=None,
+                 bias=True, **kwargs):
+
+        if border_mode not in {'valid', 'same'}:
+            raise Exception('Invalid border mode for AtrousConv2D:', border_mode)
+
+        self.output_shape_ = output_shape
+
+        super(TransposedConvolution2D, self).__init__(nb_filter, nb_row, nb_col,
+                                                      init=init, activation=activation,
+                                                      weights=weights, border_mode=border_mode,
+                                                      subsample=subsample, dim_ordering=dim_ordering,
+                                                      W_regularizer=W_regularizer, b_regularizer=b_regularizer,
+                                                      activity_regularizer=activity_regularizer,
+                                                      W_constraint=W_constraint, b_constraint=b_constraint,
+                                                      bias=bias, **kwargs)
+
+    def get_output_shape_for(self, input_shape):
+        # nb_filter = self.output_shape_[1]
+        if self.dim_ordering == 'th':
+            rows = input_shape[2]
+            cols = input_shape[3]
+        elif self.dim_ordering == 'tf':
+            rows = input_shape[1]
+            cols = input_shape[2]
+        else:
+            raise Exception('Invalid dim_ordering: ' + self.dim_ordering)
+
+        rows = conv_input_length(rows, self.nb_row,
+                                  self.border_mode, self.subsample[0])
```

Comment:
in the deconv layer, the strides actually upscale the image. I believe it will be confusing to keep "subsample".